discord_connector.py

```diff
@@ -0,0 +1,436 @@
+        status, creation_message, artifact_id = self.save_artifact(artifact)
+        self.save_progress("creating artifact: status: {}, creation message: {}, artifact id {}"
+                           .format(status, creation_message, artifact_id))
+        return artifact_id
+
+    def parse_message(self, message, attachments, embeds):
+        return {
+            "message origin": {
+                "channel id": message.channel.id,
+                "channel name": message.channel.name,
+            },
+            "message data": {
+                "created at": str(message.created_at),
+                "edited at": str(message.edited_at) if message.edited_at is not None else "message was not edited",
+            },
+            "author data": {
+                "author id": message.author.id,
+                "author name": message.author.name,
+            },
+            "jump url": message.jump_url,
+            "flags": [flag[:1] for flag in list(filter((lambda flag: flag[1]), message.flags))] or "no flags",
+            "attachments": attachments or "no attachments",
+            "embeds": embeds or "no embeds",
+            "content": message.content
+        }
+
+    def _handle_delete_message(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        channel_id = param['channel_id']
+        message_id = param['message_id']
+
+        status = self.delete_message(channel_id, message_id, action_result)
+
+        summary = action_result.update_summary({})
+        summary['action result: '] = "Deleting message {} ended with {}".format(message_id,
+                                                                                "success" if status else "failure")
+        return action_result.set_status(phantom.APP_SUCCESS) if status else action_result.set_status(phantom.APP_ERROR)
+
+    def delete_message(self, channel_id, message_id, action_result):
+        status, message = self.fetch_message(channel_id, message_id, action_result)
+        if not status:
+            return status
+        status, result = self.run_in_loop(message.delete(), action_result, error_message=DISCORD_ERROR_DELETING_MESSAGE)
+        return status
+
+    async def _load_guild(self):
+        await self._client.login(self._token)
+        self._guild = await self._client.fetch_guild(self._guild_id)
+
+    def _handle_list_channels(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        status, channels = self.run_in_loop(self._guild.fetch_channels(), action_result,
+                                            error_message=DISCORD_ERROR_FETCHING_CHANNEL)
+
+        for channel in channels:
+            if isinstance(channel, discord.TextChannel):
```

Comment:
maybe there is some way in Discord SDK to check channel type without checking objects class? It would be a bit more cleaner

Reply:
I don't see any better way to check channel type in discord sdk

Reply:
I can use
```text_channels = [channel for channel in channels if isinstance(channel, discord.TextChannel)]```
to keep text channels in seperate variable and use it to add action results

Reply:
What about https://discordpy.readthedocs.io/en/stable/api.html?highlight=textchannel#discord.TextChannel.type ?